### PR TITLE
Traffic light plugin

### PR DESCRIPTION
**Pull request by LucasJSch**
## This PR's goal is to add a traffic light to the autorace map. For this purpose several files were created and modified:

- **Traffic light plugin** (.cpp and .h files). This plugin makes the processing for changing the traffic light's color.
- **World time publisher** (.cpp and .h files). This plugin helps the first mentioned plugin to know how much time has passed since it changed the light's color.
- **Traffic light model definition** (.confing and .sdf files). In these files the visual model for the traffic light is defined.
- **autorace.world**. This file was modified for allowing to spawn the traffic light.
- **ca_gazebo's CMakeLists**. This file was modified for allowing the two new plugins to be compiled.

## To do:
- Change topic for getting the simulation time.
- Delete unnecessary world time publisher plugin.
- Resolve conflicts in branch.

## Pictures of the traffic light

red yellow green

### CODE REVIEW

**Comment by LucasJSch** [comment]
# Ready for review

File: ca_gazebo/CMakeLists.txt [hunk 1]
```diff
@@ -16,7 +16,9 @@ else()
 
   catkin_package(
 		INCLUDE_DIRS include
-		LIBRARIES create_bumper_plugin
+		LIBRARIES create_bumper_plugin 
+				  traffic_light_plugin
```
Comment (eborghi10) [inline ca_gazebo/CMakeLists.txt]: Wrong indentation.

File: ca_gazebo/CMakeLists.txt [hunk 2]
```diff
@@ -30,11 +32,21 @@ else()
 		${GAZEBO_INCLUDE_DIRS}
   )
 
-	add_library(create_bumper_plugin src/create_bumper_plugin.cpp)
+	add_library(create_bumper_plugin src/create_bumper_plugin.cpp)
```
Comment (eborghi10) [inline ca_gazebo/CMakeLists.txt]: Revert this change.

File: ca_gazebo/src/traffic_light_plugin.cpp [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+        // Subscribe to world statistics to get sim time
+        // Warning: topic ~/pose/local/info is meant for high-bandwidth local
+        // network access. It will kill the system if a remote gzclient tries to
+        // subscribe.
+        // Create our node for communication
+        this->data_ptr->node = transport::NodePtr(new transport::Node());
+        // Initialize the node with the world name
+        this->data_ptr->node->Init();
+        // Listen to Gazebo world_stats topic
+        commandSubscriber = this->data_ptr->node->Subscribe("/gazebo/EmptyWorld/WorldTime_topic", &GazeboTrafficLight::time_update_cb, this);
```
Comment (eborghi10) [inline ca_gazebo/src/traffic_light_plugin.cpp]: Why don't you get the time from `/gazebo/default/world_stats`?

File: ca_gazebo/src/traffic_light_plugin.cpp [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+
+        if(elapsed > period_time)
+        {
+            this->last_time_ = this->current_time_;
+
+            double red = 0;
+            double green = 0;
+            double blue = 0;
+
+            if(this->next_color_ == "red")
```
Comment (eborghi10) [inline ca_gazebo/src/traffic_light_plugin.cpp]: I believe this logic can be done much easier but I don't have a clue now on how to do it.

Comment (eborghi10) [reply ca_gazebo/src/traffic_light_plugin.cpp]: You can use a `std::map` that uses `this->next_color_` as key and its value is the `ignition::math::Color`.

**Review by eborghi10 — CHANGES_REQUESTED** [review]
I left some comments, thanks for pushing it!

**Comment by eborghi10** [comment]
Closing this PR because #134 fixes the conflicts with `kinetic-devel`.

**Pull request closed without merging**